packages/docsearch/index.ts

```diff
@@ -1,17 +1,37 @@
 import type { StarlightPlugin } from '@astrojs/starlight/types';
+import type docsearch from '@docsearch/js';
 import type { AstroUserConfig, ViteUserConfig } from 'astro';
 import { z } from 'astro/zod';
 
-/** Config options users must provide for DocSearch to work. */
-const DocSearchConfigSchema = z.object({
-	appId: z.string(),
-	apiKey: z.string(),
-	indexName: z.string(),
-});
-export type DocSearchConfig = z.input<typeof DocSearchConfigSchema>;
+type SearchOptions = Parameters<typeof docsearch>[0]['searchParameters'];
+
+/** DocSearch configuration options. */
+const DocSearchConfigSchema = z
+	.object({
+		// Required config without which DocSearch won’t work.
+		/** Your Algolia application ID. */
+		appId: z.string(),
+		/** Your Algolia Search API key. */
+		apiKey: z.string(),
+		/** Your Algolia index name. */
+		indexName: z.string(),
+		// Optional DocSearch component config (only the serializable properties can be included here)
+		/** The maximum number of results to display per search group. Default is `5`. */
+		maxResultsPerGroup: z.number().optional(),
+		/** Disable saving recent searches and favorites to the local storage. Default is `false`. */
+		disableUserPersonalization: z.boolean().optional(),
+		/** Whether to enable the Algolia Insights plugin and send search events to your DocSearch index. Default is `false`. */
+		insights: z.boolean().optional(),
+		/** The Algolia Search Parameters. See https://www.algolia.com/doc/api-reference/search-api-parameters/ */
```

Comment:
```suggestion
		/**
		 * The maximum number of results to display per search group.
		 * @default 5
		 */
		maxResultsPerGroup: z.number().optional(),
		/**
		 * Disable saving recent searches and favorites to the local storage.
		 * @default false
		 */
		disableUserPersonalization: z.boolean().optional(),
		/**
		 * Whether to enable the Algolia Insights plugin and send search events to your DocSearch index.
		 * @default false
		 */
		insights: z.boolean().optional(),
		/**
		 * The Algolia Search Parameters.
		 * @see https://www.algolia.com/doc/api-reference/search-api-parameters/
		 */
```

Small nit to use TSDoc tags, I personally think the highlighting of the types is more obvious as it matches the one from your code theme for that type:

| Before | After |
| --- | --- |
| <img width="340" alt="SCR-20241109-brkh" src="https://github.com/user-attachments/assets/87212286-718b-4108-bd71-6e38363f928a"> | <img width="338" alt="SCR-20241109-brup" src="https://github.com/user-attachments/assets/b574873a-e7a8-4a11-9dc3-5091d21f6613"> |

Reply:
Thanks! Yeah, I’ll admit I just copy-pasted from the Algolia docs, but agree it makes sense to use the proper tags.